src/main/java/com/minecraftabnormals/abnormals_core/core/api/model/FullbrightModelLoader.java

```diff
@@ -0,0 +1,259 @@
+            }
+
+            return list;
+        }
+    }
+
+    public static class FullbrightBlockPartFace extends BlockPartFace {
+        public boolean fullbright;
+        public boolean override;
+
+        public FullbrightBlockPartFace(@Nullable Direction cullFaceIn, int tintIndexIn, String textureIn, BlockFaceUV blockFaceUVIn, boolean fullbright, boolean override) {
+            super(cullFaceIn, tintIndexIn, textureIn, blockFaceUVIn);
+            this.fullbright = fullbright;
+            this.override = override;
+        }
+
+        public static class Deserializer extends BlockPartFace.Deserializer {
+            @Override
+            public BlockPartFace deserialize(JsonElement element, Type type, JsonDeserializationContext context) throws JsonParseException {
+                BlockPartFace face = super.deserialize(element, type, context);
+                JsonObject object = element.getAsJsonObject();
+
+                if (object.has("fullbright") && !JSONUtils.isBoolean(object, "fullbright"))
+                    throw new JsonParseException("Expected fullbright to be a Boolean");
+                return new FullbrightBlockPartFace(
+                        face.cullFace,
+                        face.tintIndex,
+                        face.texture,
+                        face.blockFaceUV,
+                        JSONUtils.getBoolean(object, "fullbright", false),
+                        object.has("fullbright") && JSONUtils.isBoolean(object, "fullbright")
+                );
+            }
+        }
+    }
+
+    public static class FullbrightBlockPart extends BlockPart {
+        public boolean fullbright;
+
+        public FullbrightBlockPart(Vector3f positionFrom, Vector3f positionTo, Map<Direction, BlockPartFace> mapFaces, @Nullable BlockPartRotation partRotation, boolean shade, boolean fullbright) {
+            super(positionFrom, positionTo, mapFaces, partRotation, shade);
+            this.fullbright = fullbright;
+        }
+
+        public static class Deserializer implements JsonDeserializer<FullbrightBlockPart> {
+        	@Override
+            public FullbrightBlockPart deserialize(JsonElement element, Type type, JsonDeserializationContext context) throws JsonParseException {
+                JsonObject object = element.getAsJsonObject();
+                Vector3f from = this.validateVectorBounds(object, "from");
+                Vector3f to = this.validateVectorBounds(object, "to");
+                BlockPartRotation rotation = this.parseRotation(object);
+                Map<Direction, BlockPartFace> faces = this.parseFacesCheck(object);
+
+                if (object.has("shade") && !JSONUtils.isBoolean(object, "shade"))
+                    throw new JsonParseException("Expected shade to be a Boolean");
+                if (object.has("fullbright") && !JSONUtils.isBoolean(object, "fullbright"))
+                    throw new JsonParseException("Expected fullbright to be a Boolean");
```

Comment:
The `JSONUtils.getBoolean()` calls below should handle these exceptions.

Reply:
They need to be there to catch if they are booleans or not there since `fullbright` and `shade` are optional and may not exist and have a fallback if they aren't actually there.